src/machines/commerce-machine.ts

```diff
@@ -22,7 +22,7 @@ const extractAppliedDefaultCoupon = (
   const appliedCoupon = pricingData?.applied_coupon
 
   // no applied default coupon found
-  if (isEmpty(availableDefaultCoupon) || isEmpty(appliedCoupon)) {
+  if (isEmpty(availableDefaultCoupon) && isEmpty(appliedCoupon)) {
```

Comment:
we still want to check below iff we have an applied coupon that isn't region restricted.